clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -881,7 +881,8 @@ class CIRAllocaLowering
                   typeConverter->convertType(rewriter.getIndexType()),
                   rewriter.getIntegerAttr(rewriter.getIndexType(), 1));
     auto elementTy = getTypeConverter()->convertType(op.getAllocaType());
-    auto resultTy = mlir::LLVM::LLVMPointerType::get(getContext());
+    auto resultTy = getTypeConverter()->convertType(op.getResult().getType());
+    // TODO: Verification between the CIR alloca AS and the one from data layout
```

Comment:
Sorry for the late push, Julian meant approving the updated patch.

Here we still plan to add a verification to ensure that there's no mismatch between the alloca AS from `TargetInfo` and the one from target data layout, similar to the idea of #672. Theoretically they are both from the specific target and should be aligned.

It requires more changes thus coming in the next PR.